src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -807,6 +818,14 @@ - (void)asyncDispatchToMatterQueue:(void (^)(chip::Controller::DeviceCommissione
     });
 }
 
+- (void)discoverCommissionableNodes
+{
+    _mDNSResolver.Shutdown(); // reset if already inited
+    _mDNSResolver.Init(chip::DeviceLayer::UDPEndPointManager());
+    _mDNSResolver.SetCommissioningDelegate(_deviceDiscoveryDelegateBridge);
+    chip::Dnssd::DiscoveryFilter filter(chip::Dnssd::DiscoveryFilterType::kNone, (uint64_t) 0);
+    _mDNSResolver.DiscoverCommissionableNodes(filter);
```

Comment:
Pretty sure this can lead to thread data races, since it's going to touch the DNS service bits on some random caller thread.

Reply:
I removed all discovery and ethernet pairing changes because they are useful for test but not a mandatory feature for Matter certification.